apps/client/src/common/components/identify-overlay/IdentifyOverlay.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Badge } from '@chakra-ui/react';
+
+import { setClientRemote } from '../../hooks/useSocket';
+import { useClientStore } from '../../stores/clientStore';
+import { socketSendJson } from '../../utils/socket';
+
+import './Overlay.scss';
+
+export default function IdentifyOverlay() {
+  const { clients, id, name, setMyName } = useClientStore();
+  const { setIdentify, setRedirect, setRename } = setClientRemote;
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (id && clients[id] && clients[id].redirect !== '') {
+      const { redirect } = clients[id];
+      if (redirect !== window.location.pathname) {
+        setRedirect({ target: id, path: '' });
+        navigate(redirect);
+      } else {
+        setRedirect({ target: id, path: '' });
+      }
+    }
+  }, [clients, id, navigate, setRedirect]);
+
+  useEffect(() => {
+    if (id && clients[id] && clients[id].rename !== '') {
+      const { rename } = clients[id];
+      if (rename !== name) {
+        socketSendJson('set-client-name', rename);
+      } else {
+        setRename({ target: id, name: '' });
+      }
+    }
+  }, [clients, id, name, setMyName, setRename]);
```

Comment:
question: I am not sure I follow the logic here.
Why would renaming so many steps? I think the request for renaming should rename the client and that is it.
No need for client involvement